### ISSUE TITLE
Switch from global variables to a builder

### ISSUE THREAD

**Issue by jstuyts** [tatsuhiro-t/argparse4j#72]
At the moment the argument parser is configured using global variables (in `ArgumentParsers` and the default locale of the JVM). When multiple instances are used on the same or other threads, configuration of a later one affects the behavior of previously instantiated parsers.

I have seen this behavior when I was modifying the locale in tests. As the tests run in parallel, the locale changes in one test thread resulted in failures in tests running on other threads.

By switching to a builder, each argument parser will get its own copy of a configuration. The proposed API is:

    ArgumentParsers.newFor("prog").addHelp(false).locale(Locale.FRENCH).[other settings].build()

When I create a PR can I assume the packages `impl` and `internal` are not part of the API? I already converted the code and I have some backwards compatible changes there.

**Comment by tatsuhiro-t** [comment]
If I understand it correctly, the issue is that we use global static variable in ArgumentParsers class and default local when creating ArgumentParserImpl.  I'm fine to make this kind of change as long as the backward compatibility is held.

impl contains classes exported for public use, so this is public API, and the backward compatibility should be taken care of.
internal is private API, so we can change its signature whatever we like.

**Comment by jstuyts** [comment]
Yes, you understood correctly. I will review the changes I have in `impl` to make sure that I don't break backward compatibility before I create a PR.

**Comment by tatsuhiro-t** [comment]
Fixed by #72
